modules/learn/pages/data/transactions.adoc

```diff
@@ -0,0 +1,56 @@
+This isolation-level is known as https://jepsen.io/consistency/models/read-committed[Read Committed].
+
+Note that this use of a document's extended attributes area results in an approximate doubling of its pre-transaction size, while the transaction is underway.
+In consequence, transactions can only be used on documents whose maximum size is 10 MB (which is half of the maximum-permitted document-size, 20 MB).
+
+As well as making use of individual documents' extended attributes, Distributed ACID Transactions also create additional documents in each bucket that they access.
+These include _Active Transaction Records_ (up to 1024 of which can exist per bucket, and whose names are prefixed with `atr-`), and _Transaction Client Records_ (one of which exists per bucket, duly named `txn-client-record`).
+These documents are automatically maintained by Couchbase Server, and should not be modified by any application.
+They have no impact on use of either the _Query Service_ or the _Search Service_.
+Note that transaction records are not written to nodes that do not contain data to be updated: consequently, only nodes that _do_ contain data to be updated are affected by a transaction.
+
+No two Distributed ACID Transactions can access the same document at the same time.
+If such an attempt occurs, only one of the transactions can proceed; while the other must await commitment of the data, before itself being able to modify it.
+
+Distributed ACID Transactions perform commits by means of _durable writes_.
+Each commit provides _eventual consistency_, which is consistent with the _single-document write_ model of durable writes; as described in xref:learn:data/durability.adoc[Durability].
+
+[#dependencies-and-limitations]
+=== Dependencies and Limitations
+
+_Distributed ACID Transactions_ constitute an evolving technology.
+Currently, the following dependencies and limitations apply:
+
+* Transactions require the Enterprise Edition 6.5 of Couchbase Server.
+They also require the Couchbase xref:3.0@java-sdk::start-using-sdk.adoc[Java SDK], version 3.0 or after.
+Additionally, it is strongly recommended that _Network Time Protocol_ (NTP) be used to synchronize time across all cluster-nodes: this ensures that abandoned transactions are cleaned up in a timely manner.
+See xref:learn:install/synchronize-clocks-using-ntp.adoc[Clock Sync with NTP].
+
+* Only documents whose initial size is 10 MB or less can be included in a transaction.
+
+* Commits are performed by means of _durable writes_, which are single-document only.
+Consequently, _eventual consistency_ is achieved as the writes occur.
+Therefore, to ensure atomicity, non-transactional updates (such as might be performed by means of the _Data Service_ or _Query Service_) should not be made to any document involved in a transaction while the transaction is itself in progress.
```

Comment:
I divide atomicity into two types (I'm going to run this by Shivani to see if she agrees):

_Write_ atomicity is making sure that either all the docs are written, or none are (rollback).  This is the 'A' in ACID, and we have this.

_Read_ atomicity is the impression (if not the reality) that a commit happens in one atomic step.  As a distributed database, we cannot do this (without destroying performance, at least).  The Jepsen paper was the first to really call out this atomic commit as not being a guaranteed thing that all databases offer, and it describes the Monotonic Atomic View (MAV) isolation level to cover it.  So read atomicity is part of the isolation level, rather than the Acidic 'A'.  We have this when two transactions interact, but not with N1QL, regular KV reads, etc.  They don't do the MAV logic.

Given that, hopefully it's clear that the issue with non-transactional updates doesn't affect either read or write atomicity.  The problem is that those updates can be lost, as the transaction will go ahead and overwrite them (once it's reached the commit point, it has to complete the commit).

Reply:
So, what do we say here? Do you want to change this? If I remove 'to ensure atomicity', is the rest of the sentence good?